module-4/backend/week_two.md

```diff
@@ -1,22 +1,71 @@
-7. What's the scope of variables in Javascript?
-8. What's the difference between `==` and `===` in JavaScript?
-9. Why do front end frameworks exist?
+1. What's one difference between ES5 and ES6?  
+
+  One difference between ES5 and ES6 is the introduction of  `const` and `let` in ES6. ES5 only had `var` when assigning variables. `const` represents constant that can be assigned to a function or a value, and it is used when something should not change in runtime. `let` allows for block scoping.  
+
+2. What's the difference between asynchronous and synchronous JavaScript?  
+
+  Synchronous runs in a single thread, where each function must wait on the prior function to finish executing before it is executed. Asynchronous JavaScript utilizes the event loop and event queue to run functions once the stack is cleared. For example, a function may be called initially that gets added to the event queue, and then the code after this function gets executed without having to wait for the initial function to complete.  
+
+3. What are the four pillars of Object Oriented programming?  
+
+  - Abstraction: this pillar focuses on using code to represent real-world ideas/objects that have behaviors (properties) and state (current values of the characteristics).
+
+  - Encapsulation: this pillar focuses on breaking down behaviors into smaller methods and only publicly exposing a larger method that is made up of the smaller, encapsulated methods. It also focuses on the idea that the behaviors of an object are defined/contained within it.
+
+  - Polymorphism: this pillar means that the same method name can be used by different objects that have this method defined according to the specifics of the object. For example, both a `car` and `bike` object might have a `tires` method that is calculated differently but uses the same method name.  
+
+  - Inheritance: this pillar allows behaviors to be defined on a parent class and then a child class gets access to these methods. An object can only inherit from a single class, but an object has access to all the methods of its parent and anything that the parent has inherited from (ancestors).
+
+4. What are some tools available in JavaScript to help you write object oriented code?  
+
+  For OOP with JavaScript, classes can be a form of encapsulation for related behaviors and represent abstraction of a real world object. We can also extend classes so that behaviors can be shared.
+
+5. What are some key concepts to be aware of when refactoring your JavaScript?  
+
+  As I'm refactoring in JavaScript, I think about it from low-level refactors to higher-level refactors. On the lower end, I look at specificity in naming, consistency in function declaration (function name() {} vs name = () => {}), and semicolon usage. On the higher-level, I look at decoupling methods to ensure modularity and single responsibility principle to allow reusability.  
```

Comment:
I love the way you walked me through your process; you've clearly put thought into this important step in development!